### ISSUE TITLE
translate-c: compile error when a float is explicitly cast to bool

### ISSUE THREAD

**Issue by f-cozzocrea** [ziglang/zig#18535]
### Zig Version

0.12.0-dev.2150+63de8a598

### Steps to Reproduce and Observed Behavior

Implicit casts from floating type to bool work, but an explicit cast from a float to a bool results in a compile error.

Reproducible example as a test case:
```c
// a.c
#include <stdbool.h>

int main() {
    float f = 2.0;
    bool b = (bool) f;
    return 0;
}

// run-translated-c
// c_frontends=clang
```

Compile error:
```
a.zig:60:29: error: expected integer type, found 'bool'
    var b: bool = @as(bool, @intFromFloat(f));
```

### Expected Behavior

Successful compilation.

**Comment by Rexicon226** [comment]
This is an interesting one, that I'd like to actually look into fixing myself @Vexu, if you don't mind. It seems simple enough?

**Comment by f-cozzocrea** [comment]
@Rexicon226 I think I have a fix, but wanted to open an issue before making a PR. Looks like it's just an oversight in `transCCast`.

**Comment by Rexicon226** [comment]
> @Rexicon226 I think I have a fix, but wanted to open an issue before making a PR. Looks like it's just an oversight in `transCCast`.

That's what I thought as well. It's fine, you can make it.

**Comment by f-cozzocrea** [comment]
Fixed in #18578.
